ComplianceSuite/Tests/RegoComplianceTests/RegoComplianceTests.swift

```diff
@@ -121,8 +123,113 @@ struct ComplianceTests {
         }
         return
     }
+    
+    private static func addTestBuiltins(to registry: BuiltinRegistry) -> BuiltinRegistry {
+        guard !registry.hasBuiltin("test.sleep") else {
+            return registry
+        }
+
+        let newRegistry = BuiltinRegistry(
+            builtins: registry.builtins.merging(
+                ["test.sleep" : TestBuiltins.test_sleep],
+                uniquingKeysWith: { (current, _) in current })
+        )
+        return newRegistry
+    }
 }
 
 extension ComplianceTesting.IRTestCase: CustomTestStringConvertible {
     public var testDescription: String { "\(description)" }
 }
+
+struct TestBuiltins {
+    static func test_sleep(ctx: BuiltinContext, args: [AST.RegoValue]) async throws -> AST.RegoValue {
+        guard args.count == 1 else {
+            throw BuiltinError.argumentCountMismatch(got: args.count, want: 1)
+        }
+
+        guard case .string(let x) = args[0] else {
+            throw BuiltinError.argumentTypeMismatch(arg: "x", got: args[0].typeName, want: "string")
+        }
+
+        guard let timeDuration = try? parseDurationNanoseconds(x), timeDuration > 0 else {
+            return .null
+        }
+
+        try await Task.sleep(nanoseconds: UInt64(timeDuration))
+
+        return .null
+    }
+
+    enum DurationParseError: Error {
+        case invalidFormat
+    }
+
+    /// Parses a duration string and returns the equivalent duration in nanoseconds.
+    ///
+    /// - Parameters:
+    ///   - duration: A string representing a duration with parts in the format of numeric values followed by a unit (e.g., "10m", "1s", "10ms").
+    ///
+    /// - Returns: The duration represented as an `Int64` in nanoseconds.
+    ///
+    /// - Throws: `DurationParseError.invalidFormat` if the input string doesn't conform to the expected format.
+    ///
+    /// - Note: Supports units such as nanoseconds (ns), microseconds (us, µs),
+    ///        milliseconds (ms), seconds (s), minutes (m), and hours (h).
+    static func parseDurationNanoseconds(_ duration: String) throws -> Int64 {
```

Comment:
⛏️ we should probably add tests for this somewhere - it's non obvious how values like `10ns10ns` are meant to behave - any it would be useful to lock that in.

Reply:
Added unit tests for this